lib/services/sorting/post/post_sorting_service.dart

```diff
@@ -25,32 +25,30 @@ class PostSortingService {
           .compareTo(option.scoreFunction(postB, position));
       // Multiplying by -1 reverses the order
       final sign = option.sortIncreasing ? 1 : -1;
+
       return sign * order;
     }
 
     late final Comparator<PostFirestore> comparator;
-    if (putOnTop.isEmpty) {
-      // This is a slight optimisation that allows the function
-      // to have the expected complexity when putOnTop is empty
-      // (though having putOnTop non empty does not change the asymptotic
-      // complexity of the function if it is a hash set/of constant size)
-      comparator = defaultComparator;
-    } else {
-      comparator = (a, b) {
-        final putOnTopA = putOnTop.contains(a.id);
-        final putOnTopB = putOnTop.contains(b.id);
-
-        if (putOnTopA && !putOnTopB) {
-          // a is considered smaller
-          return -1;
-        } else if (!putOnTopA && putOnTopB) {
-          return 1;
-        }
-
-        return defaultComparator(a, b);
-      };
+
+    int evaluateComparator(PostFirestore a, PostFirestore b) {
+      final putOnTopA = putOnTop.contains(a.id);
+      final putOnTopB = putOnTop.contains(b.id);
+
+      if (putOnTopA && !putOnTopB) {
+        // a is considered smaller
+        return -1;
+      } else if (!putOnTopA && putOnTopB) {
+        return 1;
+      }
+
+      return defaultComparator(a, b);
     }
 
+    putOnTop.isEmpty
```

Comment:
Here what about the following:

```dart
final comparator = putOnTop.isEmpty ? defaultComparator : evaluateComparator;
```

Reply:
Thanks for the suggestion! Added with [f372091](https://github.com/ProximaEPFL/proxima/pull/262/commits/f3720915f1c939cf7ed82c8b81ce2625a74b8bdc)